src/spawnUnits/main.ts

```diff
@@ -189,27 +190,46 @@ async function handleMarkChangeEvent(event: MarkChangeEvent) {
       // TODO: use the map marker to post errors back to the user
 
       const { position: markPosition } = addedMark
+      const { coalition = addedMark.coalition } = command
 
-      const accuracy = 250 // meters
+      let radius = CONST_DEFAULT_DESTROY_RADIUS_METERS
+      if (command.radius) {
+        radius = command.radius
+      }
 
-      const foundUnits = (await nearbyUnits(markPosition, accuracy))
-        .map(unit => {
-          const { lat, lon, alt } = unit
-          const unitPosition = { lat, lon, alt }
-          return { unit, distance: distanceFrom(markPosition, unitPosition) }
-        })
-        .sort((a, b) => a.distance - b.distance)
+      const foundUnits = (
+        await nearbyUnits(markPosition, radius, coalition)
+      ).map(unit => {
+        const { lat, lon, alt } = unit
+        const unitPosition = { lat, lon, alt }
+        return { unit, distance: distanceFrom(markPosition, unitPosition) }
+      })
 
       if (foundUnits.length < 1) {
         throw new Error('no nearby units found to destroy')
       }
 
-      const closestUnit = foundUnits[0].unit
-
-      const { name } = closestUnit
+      // if radius is given delete each unit inside radius
+      if (command.radius) {
+        await Promise.all(
+          foundUnits.map(async element => {
+            if (element.distance <= radius) {
+              const { name } = element.unit
+              await destroy(name)
+              await unitGone({ name })
+            }
+          })
+        )
+      }
+      // if no radius given, delete closest unit
+      else {
```

Comment:
same, 
there's no diff between one and many so we should use an array of one item and a single code path for destroys

Reply:
```
if (typeof command.radius !== 'number') {
  while (foundUnits.length > 1) { foundUnits.pop() }
}
```

Reply:
https://stackoverflow.com/a/39510882
This could also be used to only get the first item after sort
```
foundUnits.length = 1;
```